### ISSUE TITLE
OSError: [Errno 2] No such file or directory

### ISSUE THREAD

**Issue by ErickMUO** [uoregon-postlethwait/prost#2]
Hi !

I'm very new at python and I am recently trying to use PROST in my own data. However, I'm getting this error:

Traceback (most recent call last):
  File "/home/emucino/.local/lib/python2.7/site-packages/prost/command_line.py", line 54, in main_wrapper
    prost.prost.main()
  File "/home/emucino/.local/lib/python2.7/site-packages/prost/prost.py", line 5422, in main
    _alignments.align(_conf)
  File "/home/emucino/.local/lib/python2.7/site-packages/prost/alignment.py", line 127, in align
    self.genome_alignment.align(skip)
  File "/home/emucino/.local/lib/python2.7/site-packages/prost/alignment.py", line 239, in align
    self.alignment_execution.align(skip)
  File "/home/emucino/.local/lib/python2.7/site-packages/prost/alignment.py", line 481, in align
    f.write(subprocess.check_output(cline, stderr=subprocess.STDOUT))
  File "/home/emucino/.conda/envs/py2/lib/python2.7/subprocess.py", line 216, in check_output
    process = Popen(stdout=PIPE, *popenargs, **kwargs)
  File "/home/emucino/.conda/envs/py2/lib/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/home/emucino/.conda/envs/py2/lib/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
**OSError: [Errno 2] No such file or directory**

Total Prost running time: 149.3s (or 2m29s).

Any clue how I could solve it? I am working in a cluster (Debian GNU/Linux 10). I installed PROST using the documentation guide.

**Comment by jasonsydes** [comment]
Hi there @ErickMUO, Prost isn't my main project anymore, but it is indeed still in use in the Postlethwait lab and possibly other places. I can try to assist you, but just be warned that my responses may be quite delayed due to my other responsibilities.

It looks like one of your required files might not be in the correct place. (Apologies for the terrible error code.) I need more information.

Have you been able to run the [Quick Start Demo](https://prost.readthedocs.io/en/latest/demo.html) successfully?

It will help me to help you if you provide the following:
* The exact command that you ran
* Your prost.config file
* Your samples_filelist
* The "prost.log" file, if it was generated (it'll be in the same directory that you ran prost from)
* A directory listing of the directory from which you ran prost (just type "ls -al" and copy and paste the output into here)

Note: Feel free to send me redacted versions of the above, if there's any sensitive data in there (e.g., edit your sample names to something random), as this github conversation is public.

**Comment by ErickMUO** [comment]
It's Ok. Thank you for your quick response. Sorry, I didn't try the Demo I began with my own data, but I'll do it ASAP. 

1) The command I use was simply **prost** in the folder containing all the mandatory files.
2) prost.config file
[prost.config.txt](https://github.com/uoregon-postlethwait/prost/files/7908937/prost.config.txt)

3) samples_filelist (two-column file with directories to sample in 1 column and the sample name in the other)

/Dir/to/sample/Sample_Name1_R1.fastq.gz	Sample_Name1_R1
..
/Dir/to/sample/Sample_NameN_R2.fastq.gz	Sample_NameN_R1

4) Prost.log
[prost.log](https://github.com/uoregon-postlethwait/prost/files/7908939/prost.log)


5) directory listing 
-rw-r--r--  1 user 1107654500 Jun 18  2021 Homo_sapiens.GRCh38.dna.toplevel.fa.gz
drwxr-xr-x  2 user        331 Jan 20 14:46 Prost_example
-rw-r--r--  1 user          0 Jan 20 11:25 alignment_results_bbmap_genome_hsa.log
-rw-r--r--  1 user    6132877 Jan 19 14:49 hairpin.fa
-rw-r--r--  1 user    6132877 Mar 11  2018 hairpin.uracil.fa
-rw-r--r--  1 user   75076893 Jan 18 14:18 hsa_biomart_other_rnas.fasta
-rw-r--r--  1 user        792 Jan 19 14:59 list_small_Files.list
-rw-r--r--  1 user    3866196 Jan 19 14:48 mature.fa
-rw-r--r--  1 user    3866196 Mar 11  2018 mature.uracil.fa
-rw-r--r--  1 user         23 Jan 18 13:30 miRBase_release_22.1
-rw-r--r--  1 user       4850 Jan 19 14:13 prost.config
-rw-r--r--  1 user      4850 Jan 19 13:04 prost.config.example
-rw-r--r--  1 user       9167 Jan 20 11:25 prost.log
drwxr-xr-x  4 user         45 Jan 19 12:43 ref
-rw-r--r--  1 user         33 Jan 20 11:25 search.fa

**Comment by jasonsydes** [comment]
Can you type the following on the command line and tell me what it says? 

which bbmapskimmer.sh

I suspect you may not have bbmapskimmer.sh in your PATH?

**Comment by ErickMUO** [comment]
I think you are right. Nothing prompts in the terminal when i look up for bbmapskimmer.sh

**Comment by jasonsydes** [comment]
So I would still recommend that you run through the demo. In fact, it even checks to see if bbmapskimmer.sh is in your PATH, and if not, it tells you how to add it to your path.  Running through the demo will probably save you time in the end, as it may uncover and help solve other issues.

For now though, this is what the demo would've told you to do:
```
Please add the directory containing your download of BBMap to your $PATH, like so:

export PATH=$PATH:/path/to/bbmap/
```
Also, I'm remembering there's some helpful documentation on getting it setup here:
https://prost.readthedocs.io/en/latest/bbmap.html

**Comment by ErickMUO** [comment]
Thank you for your comments. We fixed a lot of errors by creating an environment with the prost requirements. Also, I ran the demo, and by doing so I noticed that the pipeline does not work properly if the input sample files(own data) are compressed. Fortunately, now it works. However now dealing with some criteria in the config.file. Additionally, may I ask if there is an easy solution to make bbmap read files as paired-end?
